packages/core/src/service/services/asset.service.ts

```diff
@@ -557,15 +560,19 @@ export class AssetService {
 
     private getHostEntityIdProperty(entity: EntityWithAssets): string {
         const entityName = entity.constructor.name;
-        switch (entityName) {
-            case 'Product':
-                return 'productId';
-            case 'ProductVariant':
-                return 'productVariantId';
-            case 'Collection':
-                return 'collectionId';
-            default:
-                throw new InternalServerError('error.could-not-find-matching-orderable-asset');
+        try {
+            switch (entityName) {
+                case 'Product':
+                    return 'productId';
+                case 'ProductVariant':
+                    return 'productVariantId';
+                case 'Collection':
+                    return 'collectionId';
+                default:
+                    return `${camelCase(entityName, true)}Id`;
+            }   
+        } catch (error) {
```

Comment:
This try-catch block is redundant, since there is no possibility of an error being thrown from the switch statement. You can omit this and just use the switch statement instead.